manifests/config.pp

```diff
@@ -10,13 +10,72 @@
     system => true,
   }
 
+  case $facts['os']['family'] {
+    'RedHat':  {
+      $nologin_shell = '/sbin/nologin'
+      case $facts['os']['release']['major'] {
+        '7': {
+          systemd::unit_file { 'caddy.service':
+            path    => '/etc/systemd/system',
+            content => template('caddy/etc/systemd/system/caddy.service.erb'),
+            enable  => true,
+            require => Class['caddy::package'],
+            notify  => Service['caddy'],
+          }
+        }
+        '6': {
+          file {'/etc/init.d/caddy':
+            ensure  => file,
+            mode    => '0744',
+            owner   => 'root',
+            group   => 'root',
+            content => template('caddy/etc/init.d/caddy.erb'),
+            require => Class['caddy::package'],
+          }
+        }
+        default: {
+          fail("${facts['os']['name']} ${facts['os']['release']['major']} is not supported.")
+        }
+      }
+    }
+    'Debian':  {
+      $nologin_shell = '/usr/sbin/nologin'
+      case $facts['os']['release']['major'] {
```

Comment:
Given all recent Debian versions run on systemd, perhaps this check can just be dropped? It's up to the user to verify it matches metadata.json.